### PR TITLE
Transform: adding missing "table"-transform and "series to rows"-transform to Grafana v7-transforms.

### LINKED ISSUE #24891: v7: Tables no longer merge on matching columns
With Grafana v7, I set up two Prometheus queries, like this:

```
sum by (cluster, region, namespace) (kube_pod_container_resource_requests_cpu_cores)
```

```
sum by (cluster, region, namespace) (kube_pod_container_resource_limits_cpu_cores)
```

In previous versions, results were merged on matching tags, which is what I wanted. Now, we have to be explicit about how data is being merged. I don't mind this, but it's extremely limited. The "outer join" transformation only allows us to join on a single label when I want to join on the three I call out in my query.

I can use `label_join` in my PromQL query to create a new label to join on. This works, but then I get joined data where I have two cluster, region, and namespace. I don't have any way to hide one of the duplicate columns.

In my opinion, this is a significant regression and makes tables unusable. Is there any plan to allow for joining time series on multiple columns?

### PR DESCRIPTION
**Pull request by mckn**
**What this PR does / why we need it**:
In version prior to Grafana v7 the Table panel had a couple of transform options that basically combined multiple time series/table data into one single result that then was visualized in the table panel.

Unfortunately, for a couple of reasons, we missed adding all of the same transforms to the new `Transform` functionality that can be used for all panels.

This PR will re-add the functionality provided by the `transform: table` (now called merge) and `transform: series to rows` (now called series to rows) in the old table panel but via the new `Transforms` functionality.

**Merge**
Will basically merge the result form two queries into one single table result. If values from the different queries share the same combined fields and don't have any conflicting values they will be merged into the same row.

Example:

Query A:

| Time                | Job     | Uptime    |
|---------------------|---------|-----------|
| 2020-07-07 11:34:20 | node    | 25260122  |
| 2020-07-07 11:24:20 | postgre | 123001233 |

Query B:

| Time                | Job     | Errors |
|---------------------|---------|--------|
| 2020-07-07 11:34:20 | node    | 15     |
| 2020-07-07 11:24:20 | postgre | 5      |

Result after `merge`-transform:

| Time                | Job     | Errors | Uptime    |
|---------------------|---------|--------|-----------|
| 2020-07-07 11:34:20 | node    | 15     | 25260122  |
| 2020-07-07 11:24:20 | postgre | 5      | 123001233 |

**Series to rows**
Will basically merge the result form two time series queries into one single table result with the following structure: `Time`, `Metric` and `Value`. The `Metric` will contain the name for the query that the value originate from.

Query A:

| Time                | Temperature |
|---------------------|-------------|
| 2020-07-07 11:34:20 | 25          |
| 2020-07-07 10:31:22 | 22          |
| 2020-07-07 09:30:05 | 19          |

Query B:

| Time                | Humidity |
|---------------------|----------|
| 2020-07-07 11:34:20 | 24       |
| 2020-07-07 10:32:20 | 29       |
| 2020-07-07 09:30:57 | 33       |

Result after `series to rows`-transform:

| Time                | Metric      | Value |
|---------------------|-------------|-------|
| 2020-07-07 11:34:20 | Temperature | 25    |
| 2020-07-07 11:34:20 | Humidity    | 22    |
| 2020-07-07 10:32:20 | Humidity    | 29    |
| 2020-07-07 10:31:22 | Temperature | 22    |
| 2020-07-07 09:30:57 | Humidity    | 33    |
| 2020-07-07 09:30:05 | Temperature | 19    |

**Which issue(s) this PR fixes**:
Fixes #24891 

**Special notes for your reviewer**:
Please have a look at the `merge` transformer. I am a bit afraid that it might not be optimal in performance when it comes to looping the data or creating the index.

### CODE REVIEW

**Comment by torkelo** [comment]
Anything remaining here?

**Comment by mckn** [comment]
> Anything remaining here?

The latest push covers it all. Took a bit more time then I expected to get everything in place with proper sorting and merging logic in place. It is ready for testing and please feel free tro start reviewing it.

I will update the docs and then change this one from draft to ready for review.

**Comment by mckn** [comment]
@oddlittlebird @torkelo what do you guys think about using tables the same way in the documentation to give some examples? Currently I am using images but it might be easier to maintain markdown tables?

**Comment by torkelo** [comment]
Using markdown tables Is a good idea

**Review by torkelo — APPROVED** [review]
Tested this some more and can't find any issues with the results! 

And did a quick read through of the code and looks solid! Great job!

**Comment by oddlittlebird** [comment]
Using markdown tables works for me. We already do that in some of the documentation.

File: docs/sources/panels/transformations.md [hunk 1]
```diff
@@ -96,7 +97,7 @@ After I apply the transformation, there is no time value and each column has bee
 
 ### Merge
 
-Use this transformation to combine the result from multiple queries into one single result based on the time field. This is helpful when using the table panel visualization. 
+Use this transformation to combine the result from multiple queries into one single result. This is helpful when using the table panel visualization. Values that is mergeable will be combined into the same row. They are considered to be mergeable if the shared fields contains the same data.
```
Comment (oddlittlebird) [inline docs/sources/panels/transformations.md:100]:
```suggestion
Use this transformation to combine the result from multiple queries into one single result. This is helpful when using the table panel visualization. Values that can be merged are combined into the same row. Values are mergeable if the shared fields contains the same data.
```

File: docs/sources/panels/transformations.md [hunk 2]
```diff
@@ -213,6 +204,19 @@ After I apply the transformation, my labels appear in the table as fields.
 
 {{< docs-imagebox img="/img/docs/transformations/labels-to-fields-after-7-0.png" class="docs-image--no-shadow" max-width= "1100px" >}}
 
+## Series to rows
```
Comment (oddlittlebird) [inline docs/sources/panels/transformations.md:207]:
```suggestion
## Series to rows

```

File: docs/sources/panels/transformations.md [hunk 3]
```diff
@@ -213,6 +204,19 @@ After I apply the transformation, my labels appear in the table as fields.
...
+Use this transformation to combine the result from multiple time series data queries into one single result. This is helpful when using the table panel visualization. 
+
+The result from this transformation will contain three columns: `Time`, `Metric` and `Value`. The `Metric` column is added so you easily can see from what query the metric originate from. You can be customized this value by defining `Label` on the source query.
```
Comment (oddlittlebird) [inline docs/sources/panels/transformations.md:210]: Are you referring to an example here? Or will this be true in all cases? The way it is written, it seems true in all cases. If that is the case, then you're good.

Comment (oddlittlebird) [inline docs/sources/panels/transformations.md:210]:
```suggestion
The result from this transformation will contain three columns: `Time`, `Metric`, and `Value`. The `Metric` column is added so you easily can see from which query the metric originates from. Customize this value by defining `Label` on the source query.
```

File: docs/sources/panels/transformations.md [hunk 4]
```diff
@@ -213,6 +204,19 @@ After I apply the transformation, my labels appear in the table as fields.
...
+
+In the example below, we are visualizing multiple queries returning time series data before applying the transformation.
```
Comment (oddlittlebird) [inline docs/sources/panels/transformations.md:212]:
```suggestion
In the example below, we apply a transformation to multiple queries returning time series data.
```

File: docs/sources/panels/transformations.md [hunk 5]
```diff
@@ -213,6 +204,19 @@ After I apply the transformation, my labels appear in the table as fields.
...
+
+{{< docs-imagebox img="/img/docs/transformations/time-series-before-merge-7-1.png" class="docs-image--no-shadow" max-width= "1100px" >}}
+
+Here is the same example after applying the merge transformation.
```
Comment (oddlittlebird) [inline docs/sources/panels/transformations.md:216]:
```suggestion
Here is the same example after applying the Series to rows transformation.
```

**Review by oddlittlebird — CHANGES_REQUESTED** [review]
Made some doc edits, generally looks good.

**Comment by oddlittlebird** [comment]
Be sure to note features that are for 7.1 beta.

File: docs/sources/panels/transformations.md (hunk 3 above)
Comment (mckn) [reply docs/sources/panels/transformations.md:210]: Yes, it is true for all cases

**Review by dprokop — APPROVED** [review]

**Pull request merged**